pkg/bpf/bpf.go

```diff
@@ -176,4 +190,102 @@ func (l *BpfLoader) Stop() {
 			return
 		}
 	}
+	cleanupMountPath()
+}
+
+func NewVersionMap(config *options.BpfConfig) *ebpf.Map {
+	var versionPath string
+	if config.AdsEnabled() {
+		versionPath = config.BpfFsPath + "/bpf_kmesh/map/"
+	} else if config.WdsEnabled() {
+		versionPath = config.BpfFsPath + "/bpf_kmesh_workload/map/"
+	}
+
+	_, err := os.Stat(versionPath)
+	if err == nil {
+		m := RecoverMap(config, versionPath)
+		if m != nil {
+			SetStartStatus(m)
+			return m
+		}
+	}
+
+	mapSpec := &ebpf.MapSpec{
+		Name:       "kmesh_version",
+		Type:       ebpf.Array,
+		KeySize:    4,
+		ValueSize:  16,
+		MaxEntries: 1,
+	}
+	m, err := ebpf.NewMap(mapSpec)
+	if err != nil {
+		log.Errorf("Create kmesh_version map failed, err is %v", err)
+	}
+
+	if err := os.MkdirAll(versionPath,
+		syscall.S_IRUSR|syscall.S_IWUSR|syscall.S_IXUSR|syscall.S_IRGRP|syscall.S_IXGRP); err != nil && !os.IsExist(err) {
+		log.Errorf("mkdir failed %v", err)
+		return nil
+	}
+
+	err = m.Pin(versionPath + "kmesh_version")
+	if err != nil {
+		log.Errorf("kmesh_version pin failed: %v", err)
+		return nil
+	}
+
+	Put(m)
+	SetKmeshStartStatus(NewStart)
+	return m
+}
+
+func Put(versionMap *ebpf.Map) {
+	key := uint32(0)
+	var value [16]byte
```

Comment:
how do you get to use 16

Reply:
git version is not too long, 
I wrote a number randomly that can cover gitversion